### PR TITLE
get 100.yaml, overview.yaml and undergrad.yaml from mathlib_docs

### PR DESCRIPTION
**Pull request by bryangingechen**
cf. [Zulip thread](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/website.20overview/near/208659351), leanprover-community/mathlib#4016, and leanprover-community/doc-gen#60.

I also updated `.gitignore` and `README.md` and also moved all the `urllib.request.urlretrieve` lines next to where the downloaded files were used in `make_site.py`.

### CODE REVIEW

File: make_site.py [hunk 1]
```diff
@@ -135,7 +135,10 @@ class HundredTheorem:
     links: Optional[Mapping[str, str]] = None
     note: Optional[str] = None
 
-urllib.request.urlretrieve('https://leanprover-community.github.io/mathlib_docs/export_db.json.gz', 'export_db.json.gz')
+urllib.request.urlretrieve(
+    'https://leanprover-community.github.io/mathlib_docs/export_db.json.gz',
+    'export_db.json.gz'
+)
```
Comment (PatrickMassot) [inline make_site.py:141]:
```suggestion
    'export_db.json.gz')
```

File: make_site.py [hunk 2]
```diff
@@ -146,6 +149,10 @@ class HundredTheorem:
 num_meta = len([d for d in decl_loc_map if decl_loc_map[d]['is_meta']])
 num_defns = len(decl_loc_map) - num_thms - num_meta
 
+urllib.request.urlretrieve(
+    'https://leanprover-community.github.io/mathlib_docs/100.yaml',
+    DATA/'100.yaml'
+)
```
Comment (PatrickMassot) [inline make_site.py:155]:
```suggestion
    DATA/'100.yaml')
```

File: make_site.py [hunk 3]
```diff
@@ -219,24 +226,21 @@ def from_node(cls, identifier: str, title: str, children, depth: int, parent: 'O
...
-    'https://leanprover-contrib.github.io/leanprover-contrib/version_history.yml',
-    DATA/'project_history.yaml'
-)
-
-urllib.request.urlretrieve(
-    'https://leanprover-contrib.github.io/leanprover-contrib/projects/projects.yml',
-    DATA/'projects.yaml'
+    'https://leanprover-community.github.io/mathlib_docs/overview.yaml',
+    DATA/'overview.yaml'
 )
```
Comment (PatrickMassot) [inline make_site.py:236]:
```suggestion
    DATA/'overview.yaml')
```

File: make_site.py [hunk 4]
```diff
@@ -219,24 +226,21 @@ def from_node(cls, identifier: str, title: str, children, depth: int, parent: 'O
...
-
 with (DATA/'overview.yaml').open('r', encoding='utf-8') as h_file:
     overviews = [Overview.from_top_level(index, title, elements) for index, (title, elements) in enumerate(yaml.safe_load(h_file).items())]
 
+urllib.request.urlretrieve(
+    'https://leanprover-community.github.io/mathlib_docs/undergrad.yaml',
+    DATA/'undergrad.yaml'
+)
```
Comment (PatrickMassot) [inline make_site.py:243]:
```suggestion
    DATA/'undergrad.yaml')
```

File: make_site.py [hunk 5]
```diff
@@ -249,15 +253,19 @@ class Project:
...
     maintainers: List[str]
-    stars: int 
+    stars: int
 
 github = Github()
 
+urllib.request.urlretrieve(
+    'https://leanprover-contrib.github.io/leanprover-contrib/projects/projects.yml',
+    DATA/'projects.yaml'
+)
```
Comment (PatrickMassot) [inline make_site.py:263]:
```suggestion
    DATA/'projects.yaml')
```

File: make_site.py [hunk 6]
```diff
@@ -266,6 +274,10 @@ class Project:
 
 projects.sort(key = lambda p: p.stars, reverse=True)
 
+urllib.request.urlretrieve(
+    'https://leanprover-contrib.github.io/leanprover-contrib/version_history.yml',
+    DATA/'project_history.yaml'
+)
```
Comment (PatrickMassot) [inline make_site.py:280]:
```suggestion
    DATA/'project_history.yaml')
```

File: templates/mathlib-overview.html [hunk 7]
```diff
@@ -17,17 +17,17 @@ <h1>A mathlib overview</h1>
...
 	<a href="undergrad_todo.html">not yet in mathlib</a>.
 	</p>
 
 	<p>
-	To make updates to this list, please make a pull request to this repository
-	after editing the 
-	<a href="https://github.com/leanprover-community/leanprover-community.github.io/blob/newsite/data/overview.yaml">yaml source file</a>.
+	To make updates to this list, please <a href="contribute/index.html">make a pull request to mathlib</a>
+	after editing the
+	<a href="https://github.com/leanprover-community/mathlib/blob/master/docs/overview.yaml">yaml source file</a>.
```
Comment (PatrickMassot) [inline templates/mathlib-overview.html:30]: I would also point out that PRs to yaml files can be done entirely online on GitHub (hopefully there is some GitHub help page we can link to).

**Review by PatrickMassot — COMMENTED** [review]
Thanks! I don't like all those lonely parentheses, I hope I'm not the author of any of those.

File: templates/mathlib-overview.html (hunk 7 above)
Comment (bryangingechen) [reply templates/mathlib-overview.html:30]: Good idea, thanks!

**Pull request merged**